src/miral/basic_window_manager.cpp

```diff
@@ -2902,15 +2932,25 @@ void miral::BasicWindowManager::update_application_zones_and_attached_windows()
             if (window)
             {
                 auto& info = info_for(window);
+                if (surface_stack->screen_is_locked() && !info.visible_on_lock_screen())
+                    continue;
```

Comment:
This looks wrong. BWM doesn't seem the right place for logic choosing which windows are visible in the scene.

Reply:
To be honest, a reasonable way around this is to set `ignore_exclusion_zones` to `true` for all ext-session-lock-v1 surfaces. This _would_ make sense WRT to the spec, and then we wouldn't need to update when exclusion zones change at all. This might be a better thing to do. WDYT

Reply:
@AlanGriffiths I left a question for you here, which is why I'm requesting a rereview